nifti2/nifti2_io.c

```diff
@@ -3540,7 +3540,7 @@ char * nifti_find_file_extension( const char * name )
                  ext);
          return NULL;
       }
-      else return (char *)ext; /* Cast away the constness of the input parameter */
+      else return ext;
```

Comment:
So much better!

Reply:
I don't think this will break many clients, and it's just so wrong it's worth fixing IMHO.  It's also the *only* -Wconst-qual warning left.

Reply:
There is only one other exported "const char *" function in nifti2_io.c.  This will force calling code to deal with the viral const.  The strstr method (for example) seems nicer (`man strstr`), where the library allows one to use either const or not.  I do not really see this as wrong.

Reply:
I'll move these const changes to a new PR for fresh arguments :)